### PR TITLE
[Snyk] reach lib bump 0.15

### PR DESCRIPTION
**Pull request by alexmylonas**
### Background

We want to bump @reach library to v0.15

### Changes

- bump package.json
- Updated changed types for reach utils

### Testing

- Functionality test locally

### CODE REVIEW

File: src/components/Dropdown/DropdownLink.tsx
Comment (3nvi) [inline src/components/Dropdown/DropdownLink.tsx:7]:
```suggestion
export const DropdownLink: Polymorphic.ForwardRefComponent<'div', DropdownLinkProps & { disabled?: boolean }> = ({
```

File: src/components/Popover/PopoverTrigger.tsx
Comment (3nvi) [inline src/components/Popover/PopoverTrigger.tsx:22]: we still need to wrap them. The new method is called `composeEventHandlers` on the same package

File: src/components/Popover/PopoverTrigger.tsx
Comment (3nvi) [inline src/components/Popover/PopoverTrigger.tsx:11]: Let's apply thee same changes as the ones above

**Review by 3nvi — CHANGES_REQUESTED** [review]
hey @alexmylonas , we need to make some changes to make all of this work. Before we even merge it, we need to install it in Panther and make sure that it passes our `npm run tsc` check.

You can see how you intsall a pounce branch to a project [here](https://medium.com/@jonchurch/use-github-branch-as-dependency-in-package-json-5eb609c81f1a)

File: src/components/Popover/PopoverTrigger.tsx
Comment (alexmylonas) [reply src/components/Popover/PopoverTrigger.tsx:22]: Good callout

File: src/components/Dropdown/DropdownButton.tsx
Comment (alexmylonas) [reply src/components/Dropdown/DropdownButton.tsx]: Done!

File: src/components/Dropdown/DropdownItem.tsx
Comment (alexmylonas) [reply src/components/Dropdown/DropdownItem.tsx]: Updated!

File: src/components/Dropdown/DropdownLink.tsx
Comment (alexmylonas) [reply src/components/Dropdown/DropdownLink.tsx]: Updated!

**Comment by alexmylonas** [comment]
I have installed branch on my local env. Related components seems to work & look as before!

**Review by 3nvi — APPROVED** [review]
If panther has no TS issues, then it LGTM 🎉

**Pull request merged**
